src/store/store.js

```diff
@@ -8,24 +8,32 @@ const myMovieList = new MovieListService()
 
 export const store = new Vuex.Store({
   state: {
-		movieList: {}
+		movieList: {},
+		errorMovieAPI: false
 	},
 	getters: {
 		getShows: state => {
 			return state.movieList.shows;
-		}
+		},
+		getTrailer: state => id => {
+        	const show = state.movieList.shows.filter((item) => item.imdbID.toLowerCase().indexOf(id) >= 0);
+        	return 'https://www.youtube-nocookie.com/embed/' + show[0].trailer
+		},
+		isErrorMovieAPI: state => state.errorMovieAPI
 	},
 	mutations: {
 		setMovieList: (state, responseObj) => {
 			state.movieList = responseObj;
 		} 
 	},
 	actions: {
-		setMovieList: async({commit}) => {
-			const responseObj = await myMovieList.getMovieList()
-        .then(response => response.data)
-        .catch(err => console.log(err))
-			commit("setMovieList", responseObj);	
+		setMovieList: async({commit,state}) => {
+			try {
+				const responseObj = await myMovieList.getMovieList().then(response => response.data)
```

Comment:
either you can use async await or promise then.
`
const responseObj = await myMovieList.getMovieList();
`
`commit("setMovieList", responseObj.data);`